### PR TITLE
Integration testing and fix n+1 problems

### PR DESCRIPTION
**Pull request by addisalemseifu**
- Implement eager-loading to fix N+1 problems.
- Used Capybara to write integration tests for each view in my project.
# **before**

Screenshot 2023-10-03 125332

# **after**

Screenshot 2023-10-03 125628

### CODE REVIEW

File: spec/integration/users/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    # it 'shows all the posts of the user' do
+    #     click_link('See all posts')
+    #     visit "/users/#{@user.id}/posts/"
+    #   @user.posts.each do |post|
+    #     expect(page).to have_content(post.title)
+    #     expect(page).to have_content(post.text)
+    #   end
+    # end
```
Comment (OyePriscilla) [inline spec/integration/users/show_spec.rb]: - [x] Please kindly remove this commented lines of code in order to make your codebase to be cleaner. **Note: Kindly do the same for line 50 in this file and line 44-47 in `posts/index_spec.rb` file** ℹ️

File: spec/integration/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                        photo: 'https://media.istockphoto.com/id/1406197730/photo/portrait-of-a-young-handsome-indian-man.webp?b=1&s=170667a&w=0&k=20&c=KtmKHyOE6MJV0w2DiGX8P4399KHNbZ3p8lCjTEabGcY=',
+                        bio: 'Full-Stack Software developer', posts_counter: 0)
+    @user.posts.create(title: 'Post 1', text: 'This is the first post.', comments_counter: 0, likes_counter: 0)
+    @user.posts.create(title: 'Post 2', text: 'This is the second post.', comments_counter: 0, likes_counter: 0)
+    @user.posts.create(title: 'Post 3', text: 'This is the third post.', comments_counter: 0, likes_counter: 0)
+  end
+
+  describe 'User show page' do
+    before(:each) { visit user_path(id: @user.id) }
+
```
Comment (OyePriscilla) [inline spec/integration/users/show_spec.rb:15]: - [x] Kindly implement these two missing tests as highlighted in the screenshot below and as stated in the project requirement. 🙏 

Untitled

File: spec/integration/posts/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                        bio: 'Full-Stack Software developer', posts_counter: 0)
+    @post1 = @user.posts.create(title: 'Post 1', text: 'This is the first post.', comments_counter: 0,
+                                likes_counter: 0)
+    @post2 = @user.posts.create(title: 'Post 2', text: 'This is the second post.', comments_counter: 0,
+                                likes_counter: 0)
+    @post3 = @user.posts.create(title: 'Post 3', text: 'This is the third post.', comments_counter: 0,
+                                likes_counter: 0)
+    Comment.create(post: @post1, author: @user, text: 'This is a comment')
+    Like.create(post: @post1, author: @user)
+  end
```
Comment (OyePriscilla) [inline spec/integration/posts/index_spec.rb]: - [x]  Kindly implement these two missing tests as highlighted in the screenshot below and as stated in the project requirement. 🙏

Untitled

File: spec/integration/posts/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+RSpec.describe 'Posts Show', type: :system do
+  before(:each) do
+    @user = User.create(name: 'Addisalem Seifu',
+                        photo: 'https://media.istockphoto.com/id/1406197730/photo/portrait-of-a-young-handsome-indian-man.webp?b=1&s=170667a&w=0&k=20&c=KtmKHyOE6MJV0w2DiGX8P4399KHNbZ3p8lCjTEabGcY=',
+                        bio: 'Full-Stack Software developer', posts_counter: 0)
+    @post1 = @user.posts.create(title: 'Post 1', text: 'This is the first post.', comments_counter: 0,
+                                likes_counter: 0)
+    @post2 = @user.posts.create(title: 'Post 2', text: 'This is the second post.', comments_counter: 0,
+                                likes_counter: 0)
+    @post3 = @user.posts.create(title: 'Post 3', text: 'This is the third post.', comments_counter: 0,
```
Comment (OyePriscilla) [inline spec/integration/posts/show_spec.rb:11]: - [x]  Kindly implement this missing tests as highlighted in the screenshot below and as stated in the project requirement. 🙏

Untitled

File: README.md [hunk 5]
```diff
@@ -103,12 +105,18 @@ it will install the required gemfile for running the project
...
+  👤 **addisalem**
 
 - GitHub: [@addisalemseifu](https://github.com/githubhandle)
 - Twitter: [@Addisalem Seifu](https://twitter.com/addisalems92518)
 - LinkedIn: [addisalem seifu](https://www.linkedin.com/in/addisalemseifu/)
 
+👤 **Zohra Neda**
+
+- GitHub: [@zohra-neda](https://github.com/zohra-neda)
+- Twitter: [@zohra_neda](https://twitter.com/zohra_neda)
```
Comment (OyePriscilla) [inline README.md]: - [x] Please kindly ensure that all the social media handles are accessible and redirected to the expected pages. At the moment the github account for @addisalemseifu is a wrong one I guess, the twitter accounts for @Zohra-Neda does not exist; and that of @addisalemseifu is suspended. Kindly fix this in order to make your README file professional. **Hint: If the twitter accounts are not available at the moment, I think you can remove them** 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @addisalemseifu 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented integration tests with Capybara gem ✔️ 
Secure app from n+1 problems ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 5 above)
Comment (addisalemseifu) [reply README.md]: My partner dod not have a working twitter account for now. so I just removed it.

**Review by Ibizugbe — APPROVED** [review]
Hi @Team👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] All test cases requested by the previous reviewer has been added👍🏼.
- [x] broken links in the readne file has been fixed👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**